src/viur/core/i18n.py

```diff
@@ -442,11 +476,11 @@ def add_missing_translation(
 
     # Ensure lowercase key
     key = key.lower()
-    entity = db.Query(KINDNAME).filter("tr_key =", key).getEntry()
-    if entity is not None:
+
+    # Check if key already exists
+    if db.Get(db.Key(KINDNAME, key)):
```

Comment:
Because existing entities would still keep their numeric id as key, this would not work

```suggestion    
    entity = db.Query(KINDNAME).filter("name =", key).getEntry()
    if entity is not None:
```